test/functional/fixtures/api/es-next/request-hooks/testcafe-fixtures/add-remove-request-hook.js

```diff
@@ -0,0 +1,66 @@
+import { RequestHook, ResultPromise } from 'testcafe';
+import { RequestFilterRule } from 'testcafe-hammerhead';
+
+const pageUrl = 'http://localhost:3000/fixtures/api/es-next/request-hooks/pages/index.html';
+
+class TestRequestHook extends RequestHook {
+    constructor (name) {
+        super();
+
+        this.name              = name;
+        this.requestFilterRule = new RequestFilterRule(pageUrl);
+
+        this.onResponseCallCountInternal = 0;
+
+        Object.defineProperty(this, 'onResponseCallCount', {
```

Comment:
You can use ES6 [class getter](http://2ality.com/2015/02/es6-classes-final.html#getters-and-setters) here. Also, IMHO it's better to name it `responseHandlerCallCount` (and use `internalResponseHandlerCallCount` for the internal property).

Reply:
> You can use ES6 class getter here
Ok. Let it be.

> Also, IMHO it's better to name it responseHandlerCallCount (and use internalResponseHandlerCallCount for the internal property).

It is not handler. It is method from base class.  Simular as for [Reporter plugin](https://devexpress.github.io/testcafe/documentation/extending-testcafe/reporter-plugin/#implementing-the-reporter).

Variants: `onResponseMethodCallCount`, `responseCallCount` and etc.
So, I will keep it as is.